File: backend/app/api/routes/export.py
from fastapi import APIRouter, Depends, HTTPException, Query
from typing import Dict, Any
# REMOVED CIRCULAR IMPORT: from app.api.routes.auth import get_current_user
from app.models.user import User
from ..deps import get_db, get_current_user, get_current_active_user, get_admin_user

router = APIRouter(prefix="/api/export", tags=["export"])

@router.post("/create")
async def create_export(export_data: Dict[str, Any], current_user: User = Depends(get_current_user)):
    return {"export_id": "exp-123", "status": "processing"}

@router.get("/jobs")
async def get_export_jobs(current_user: User = Depends(get_current_user)):
    return {"jobs": [], "total": 0}

@router.post("/quick-export")
async def quick_export(export_data: Dict[str, Any], format: str = Query("csv"), current_user: User = Depends(get_current_user)):
    return {"message": "Export completed", "format": format}

File: frontend/src/services/export.js
// frontend/src/services/export.js
import api from './api';

export const exportService = {
  // Create Export
  createExport: async (exportData) => {
    const response = await api.post('/api/api/export/create', exportData);
    return response.data;
  },

  // Get Export Jobs
  getExportJobs: async (params = {}) => {
    const response = await api.get('/api/api/export/jobs', { params });
    return response.data;
  },

  // Download Export
  downloadExport: async (exportId) => {
    const response = await api.get(`/api/export/download/${exportId}`, {
      responseType: 'blob'
    });
    return response.data;
  },

  // Quick Export
  quickExport: async (exportData, format = 'csv') => {
    const response = await api.post('/api/api/export/quick-export', exportData, {
      params: { format },
      responseType: 'blob'
    });
    return response.data;
  },

  // Get Export Templates
  getExportTemplates: async (category = null) => {
    const response = await api.get('/api/api/export/templates', {
      params: { category }
    });
    return response.data;
  },

  // Delete Export Job
  deleteExportJob: async (exportId) => {
    const response = await api.delete(`/api/export/jobs/${exportId}`);
    return response.data;
  }
};
